api/src/main/java/me/lokka30/treasury/api/common/NamespacedKey.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * This file is/was part of Treasury. To read more information about Treasury such as its licensing, see <https://github.com/ArcanePlugins/Treasury>.
+ */
+
+package me.lokka30.treasury.api.common;
+
+import java.util.Locale;
+import java.util.Objects;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Represents a namespaced key, used for unique identification of any so-called "non-player"
+ * object the Treasury APIs have.
+ *
+ * @author MrIvanPlays
+ * @since 2.0.0
+ */
+public final class NamespacedKey {
+
+    /**
+     * Creates a new {@code NamespacedKey}.
+     *
+     * @param namespace namespace
+     * @param key       key
+     * @return new namespaced key
+     * @throws IllegalArgumentException if the inputted namespace or key contain any spaces or if
+     *                                  the namespace is "treasury". The
+     *                                  "treasury" namespace is reserved for the Treasury plugin.
+     */
+    @NotNull
+    public static NamespacedKey of(@NotNull String namespace, @NotNull String key) {
+        Objects.requireNonNull(namespace, "namespace");
+        Objects.requireNonNull(key, "key");
+        validateNotTreasuryNamespace(namespace);
+        validateWhitespaces(namespace, "Namespaces cannot contain spaces");
+        validateWhitespaces(key, "Keys cannot contain spaces");
+        return new NamespacedKey(namespace, key);
+    }
+
+    /**
+     * Creates a new {@code NamespacedKey} via the specified {@code namespacedKey} {@link String}
+     * , formatted in a way of such: "(namespace):(key)"
+     *
+     * @param namespacedKey namespaced key
+     * @return new namespaced key
+     * @throws IllegalArgumentException if the inputted string is not correctly formatted, if the
+     *                                  inputted string contains any whitespaces, or if
+     *                                  the namespace is "treasury". The "treasury" namespace is
+     *                                  reserved for the Treasury plugin.
+     */
+    @NotNull
+    public static NamespacedKey fromString(@NotNull String namespacedKey) {
+        Objects.requireNonNull(namespacedKey, "namespacedKey");
+        int columnCharOccurrences = 0;
+        for (char c : namespacedKey.toCharArray()) {
+            if (Character.isWhitespace(c)) {
+                throw new IllegalArgumentException("A namespaced key string cannot contain spaces");
+            }
+            if (c == ':') {
+                columnCharOccurrences++;
+                if (columnCharOccurrences > 1) {
+                    break;
+                }
+            }
```

Comment:
I wonder if it would be clear if we moved this check down just below where we do the string split and check that `parts.length == 2`